raft/fsm.hh

```diff
@@ -383,12 +386,18 @@ public:
         }
     }
 
+    // Ask to search for a leader if one is not known.
+    void ping_leader() {
+        assert(!current_leader());
```

Comment:
on_internal_error()?

Maybe this assert() is not really necessary, the pinger can check for leader and do nothing.

Reply:
Interestingly, this assert() helped us found a bug which lead to creating commit 5308a7d7a35074d627b4144281bf029e6d8c3738.